File: web/src/types/resumeProfile.ts
/**
 * 声明可复用简历画像库的列表、编辑和默认项契约。
 */
export interface ResumeProfile {
  resume_profile_id: string;
  user_id: string;
  source_session_id: string | null;
  source_confirmed_profile_id: string | null;
  name: string;
  summary: string;
  target_roles: string[];
  target_directions: string[];
  core_skills: string[];
  supporting_skills: string[];
  search_keywords: string[];
  preferred_locations: string[];
  work_arrangements: string[];
  strengths: string[];
  risks: string[];
  profile: Record<string, unknown>;
  raw_resume_text: string | null;
  is_default: boolean;
  archived_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface ResumeProfileListResponse {
  items: ResumeProfile[];
}

export interface ResumeProfileUpdatePayload {
  name?: string | null;
  summary?: string | null;
  target_roles?: string[] | null;
  target_directions?: string[] | null;
  core_skills?: string[] | null;
  supporting_skills?: string[] | null;
  search_keywords?: string[] | null;
  preferred_locations?: string[] | null;
  work_arrangements?: string[] | null;
  strengths?: string[] | null;
  risks?: string[] | null;
  raw_resume_text?: string | null;
}

File: app/schemas/resume_profile.py
"""定义可复用简历画像在 API、领域服务和 JSON 快照之间共用的 Pydantic 契约。"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class ResumeProfile(BaseModel):
    resume_profile_id: str
    user_id: str
    source_session_id: str | None = None
    source_confirmed_profile_id: str | None = None
    name: str
    summary: str
    target_roles: list[str] = Field(default_factory=list)
    target_directions: list[str] = Field(default_factory=list)
    core_skills: list[str] = Field(default_factory=list)
    supporting_skills: list[str] = Field(default_factory=list)
    search_keywords: list[str] = Field(default_factory=list)
    preferred_locations: list[str] = Field(default_factory=list)
    work_arrangements: list[str] = Field(default_factory=list)
    strengths: list[str] = Field(default_factory=list)
    risks: list[str] = Field(default_factory=list)
    profile: dict[str, object] = Field(default_factory=dict)
    raw_resume_text: str | None = None
    is_default: bool = False
    archived_at: datetime | None = None
    created_at: datetime
    updated_at: datetime


class ResumeProfileListResponse(BaseModel):
    items: list[ResumeProfile] = Field(default_factory=list)


class ResumeProfileUpdateRequest(BaseModel):
    """描述画像update的输入结构。"""
    name: str | None = None
    summary: str | None = None
    target_roles: list[str] | None = None
    target_directions: list[str] | None = None
    core_skills: list[str] | None = None
    supporting_skills: list[str] | None = None
    search_keywords: list[str] | None = None
    preferred_locations: list[str] | None = None
    work_arrangements: list[str] | None = None
    strengths: list[str] | None = None
    risks: list[str] | None = None
    raw_resume_text: str | None = None
